### DISCUSSION TITLE
payload-openapi: Schema and component names

### DISCUSSION THREAD

**Discussion by barthy-koeln** [teunmooij/payload-swagger#56] [General]
Hi Teun, thanks for the great work.

I'd personally prefer using the singluar/plural names respectively for schema/component definitions.
The easiest example: The default `User` collection.

Currently it generates schemas called `users`, representing a single user, and `userss`, representing a list of users.

Using the defined Labels, it would for example be `User` and `Users`. 

I use the OpenAPI spec to generate TypeScript interfaces and services and I'd like them to ideally look as close to the payload generated types as possible. Best case scenario, they would be completely interchangeable.

Was this a concious choice, and if yes why?
If not, I would happily submit a pull request.

Barthy

**Comment by teunmooij** [comment]
Hi, for the schema names I chose to use the `slug`, because with that I'm certain that it's a valid schema name. The labels might contain spaces and special characters.
I've created a PR last week on the payload repo to allow adding custom data to the payload config. Once that's merged I'll be happy to include a way to override the schema names.

**Comment by teunmooij** [comment]
Hi @barthy-koeln,

I've just released `payload-openapi` version 1.1.1. It allows you to override the schema names by setting a `label.singular.openapi` and/or `label.plural.openapi` value.
